### ISSUE TITLE
cannot open gtf file with refGene in file name

### ISSUE THREAD

**Issue by jayoung** [igvteam/igv#588]
hi there,

I've found a bug that's quite similar to one that was fixed a while back
https://github.com/igvteam/igv/issues/212
I'm using IGV v2.4.14 (08/14/2018 04:19 PM).   

I'm trying to load a gtf file.    It loads fine if the file is called test.gtf, but I rename that file to  dm6refGene.gtf I get this cryptic error:

Error loading /Volumes/fh/fast/malik_h/grp/public_databases/UCSC/fly_Aug2014_dm6/misc_tracks/test/dm6refGene.gtf: Error parsing line at byte position: htsjdk.tribble.readers.LineIteratorImpl@5375e8da, for input source: /Volumes/fh/fast/malik_h/grp/public_databases/UCSC/fly_Aug2014_dm6/misc_tracks/test/dm6refGene.gtf

Even just changing one letter of the file name so that it doesn't contain "refGene" seems to fix it (e.g. dm6defGene.gtf loads fine)

thanks very much,

Janet

**Comment by jrobinso** [comment]
Yes, that's a known issue.  Since downloaded refGene files from UCSC don't
have distinctive extensions,  it looks for refGene in the filename and if
found assumes its that format.  The workaround is as you suggest, don't
have refGene in the name.   Although it might seem to be a simple thing to
fix that,  after 8 years of being stable changing the rule would possibly
break a large number of sessions.    However,  we could look for recognized
extension first (.gtf),  then resort to the refGene search if that fails.

**Comment by jayoung** [comment]
I see.  Yes, I can see why you don't want to break existing sessions.   

Looking for the extension first does seem like a useful fix.   

Also, is it possible to make a more informative error message?  I spent quite a while trying to figure out how my file might be formatted wrongly (that's usually my issue) before I realised that it was the file name that was the problem rather than the contents. That was really unintuitive.   Maybe it's possible to make a special error message just for files called *refGene* ?

thanks,

Janet

**Comment by jrobinso** [comment]
Yes, a special error message is a good idea.    I've noted both for my next round of IGV improvements.
